src/main/java/org/ibp/api/rest/labelprinting/StudyEntriesLabelPrinting.java

```diff
@@ -0,0 +1,261 @@
+package org.ibp.api.rest.labelprinting;
+
+import com.google.common.collect.Maps;
+import org.apache.commons.lang3.StringUtils;
+import org.generationcp.commons.util.FileNameGenerator;
+import org.generationcp.commons.util.FileUtils;
+import org.generationcp.middleware.api.brapi.v1.attribute.AttributeDTO;
+import org.generationcp.middleware.api.germplasm.GermplasmAttributeService;
+import org.generationcp.middleware.api.germplasm.GermplasmNameService;
+import org.generationcp.middleware.api.nametype.GermplasmNameTypeService;
+import org.generationcp.middleware.domain.etl.MeasurementVariable;
+import org.generationcp.middleware.domain.etl.StudyDetails;
+import org.generationcp.middleware.domain.germplasm.GermplasmNameDto;
+import org.generationcp.middleware.domain.oms.TermId;
+import org.generationcp.middleware.domain.ontology.VariableType;
+import org.generationcp.middleware.domain.study.StudyEntrySearchDto;
+import org.generationcp.middleware.manager.api.StudyDataManager;
+import org.generationcp.middleware.service.api.study.StudyEntryDto;
+import org.generationcp.middleware.service.api.study.StudyEntryPropertyData;
+import org.ibp.api.java.study.StudyEntryService;
+import org.ibp.api.rest.common.FileType;
+import org.ibp.api.rest.labelprinting.domain.Field;
+import org.ibp.api.rest.labelprinting.domain.LabelType;
+import org.ibp.api.rest.labelprinting.domain.LabelsData;
+import org.ibp.api.rest.labelprinting.domain.LabelsGeneratorInput;
+import org.ibp.api.rest.labelprinting.domain.LabelsInfoInput;
+import org.ibp.api.rest.labelprinting.domain.LabelsNeededSummary;
+import org.ibp.api.rest.labelprinting.domain.LabelsNeededSummaryResponse;
+import org.ibp.api.rest.labelprinting.domain.OriginResourceMetadata;
+import org.ibp.api.rest.labelprinting.domain.SortableFieldDto;
+import org.ibp.api.rest.preset.domain.LabelPrintingPresetDTO;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+@Transactional
+public class StudyEntriesLabelPrinting extends LabelPrintingStrategy {
+
+	public static final int ATTRIBUTE_DISPLAY_MAX_LENGTH = 200;
+
+	public static final int NAME_DISPLAY_MAX_LENGTH = 200;
+
+	protected static final List<FileType> SUPPORTED_FILE_TYPES = Arrays.asList(FileType.CSV, FileType.PDF, FileType.XLS);
+
+	public static final String ORIG_FINAL_NAME = "entries";
+
+	@Autowired
```

Comment:
@cuenyad why the services are not private?

Reply:
done.